### DISCUSSION TITLE
About the use of cutting or increasing the image area in character training

### DISCUSSION THREAD

**Discussion by DuskXi** [d8ahazard/sd_dreambooth_extension#481] [General]
I'm curious why in so many tutorials, people are taught to crop the images in the dataset instead of completing the photo to a square by adding pixels with an Alpha value of 0. This is because many anime illustrations can actually only take care of the head and not the legs when intercepting the square, thus resulting in most of the data in the dataset being only the upper half of the body.

Therefore, I tried a way, to start with use this project: `https://github.com/SkyTNT/anime-segmentation/` to preprocess the illustration by subtracting the background, and then use a simple Python script to complete the image as a square. My last model trained using this method performed well (especially for squares and vertical images, but it was flawed for horizontal images). Then I used webui's training tab's preprocessing to generate a batch of cropped pattern datasets on the same deducted background dataset before training, and the results were not as good as the model trained using the filled pattern dataset.

My training environment is based on anything v3 7G as the main model, and the original dataset has about 170 images, without flip mirroring. I am not sure if it is beneficial for me to stack so many training sets, because I rarely use less than 100 data sets for training. The anime characters I trained are game characters, and the most important purpose is to train the face, appearance and temperament of the characters.

**Comment by IVPhantom** [comment]
Hey, while you said you tried this on anything v3, have you used this method with real life images, if so, any good for training result wise? thanks

**Reply by DuskXi** [reply]
I have not used this method for training realistic pictures for the time being, but I would generally assume that this approach should be applied to cases where the person is trained and there is a need for full body input, since the screenshots do not cover the whole person, at least in many vertical pictures. Secondly another condition is that you need to have a keying AI program for realistic characters (I'm sure there's something like that on Github, although I haven't searched for it yet) for the added part of the pixel to look less out of place after keying.

**Comment by cerega66** [comment]
Dreambooth considers such things as part of the concept. But also isolates them as a separate token. It's not just their presence that matters. If all the stripes along the edges have the same size, this will be strongly written into the concept, if different, then weakly. I myself have trained concepts with more than 300 frames and they all had stripes around the edges. Sometimes they still appeared during generation, but they were removed by adding the "cropped" token to the negative.

**Reply by DuskXi** [reply]
Since I used AI to deduct the background of the image, leaving only the people, there won't be any border in my image, basically the edges of the original image are pixels with an Alpha value of 0. So after the image is added to the square, you will only see a figure in the transparent background, not a strange border or stripe. If there are stripes, it may be that your algorithm is wrong. 

When I was writing the code for enlargement, I chose the fill rgba(255,255,255,0) at the beginning, and then found that there were stripes, after that I changed the fill to rgba(0,0,0,0), so that the picture would not have stripes.

**Reply by cerega66** [reply]
The background is an important part of the information load and diversity. If you do not have it, you will very quickly go into noise (chaos), since all your images will contain only one and the same information only in different variations. If you're using class images this will reduce the effect, but I still don't recommend completely removing the background. After a training, always check for overtraining by plotting a CFG from 1 to 30.

**Reply by DuskXi** [reply]
I am curious as to what form model overfitting will take in the CFG X/Y Plot.

**Reply by cerega66** [reply]
I only have a screenshot of the noise. There is still a glut, the image is visible there, but the details and colors seem to be saturated.
![1](https://user-images.githubusercontent.com/117596277/206900565-c8261b27-3b64-4e90-840a-0ee18870c875.PNG)

**Reply by DuskXi** [reply]
Obviously you are correct.

After my three-day experiment, I proved that your statement is correct.

First of all and regardless of the results of XY Plot.

Your statement is correct in terms of character performance and background damage alone. I trained 3 models in these 3 days, using the control variables method, making sure all parameters are the same. I trained three models based on AnythingV3 as the initial training ckpt, background deduction + background fill, background deduction + clipping, and no background deduction + clipping, and the first two obviously have serious problems, that is, the background is heavily polluted, and at more than 1w steps In more than 1w steps, there will be serious background distortion, loss of details and other problems. The latest model is the one trained without background deduction, and the background is basically not very disturbing (still there, if compared with the original AnythingV3).

In addition, there is another problem common to the first two models is that the horizontal chart crashes and the characters don't like to stand up in the horizontal chart and always prefer to lie down.

I am still exploring how to preserve the original drawing performance of AnythingV3 as much as possible in the process of training DB, just purely training my character into it.

Do you have any information or ideas you can tell me? For example, how to make sure DB training doesn't overly damage the base model, or ideas for model merging (using that checkpoint merge function in webui, I'm using the add diff method to merge 3 models because the trained model is 4G and can't be merged directly with the trimmed 3.7G and full size 7G models, so a third auxiliary model is needed)

I'm curious

**Reply by cerega66** [reply]
What do you mean by "horizontal chart".
Image generation?
Or the look of the original image?

If the latter, then here is a quote from my research that I am doing:

When you call your concept on your token, it will automatically include other tokens that were encountered while training the concept, unless you specify other tokens.
If, in addition to your token, in instance_prompt you specify other tokens that are found in your concept, then calling them for the concept will become optional.
For example, you are training a character with black hair and if your instance_prompt looks like this:
"instance_prompt": "myperson black hair"
In this case, your concept will not give you a character with strictly black hair until you specify it.

If all of your images contain a laying character, then the concept will try hard to replicate that.

**Comment by chenqihui** [comment]
这是来自QQ邮箱自动回复邮件。
 
您好，我已经收到您的邮件。我将尽快给您回复。
